plugins/generator-1.21.1/neoforge-1.21.1/templates/livingentity/livingentity.java.ftl

```diff
@@ -976,5 +1027,70 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
 		return builder;
 	}
 
+	<#if data.sensitiveToVibration>
+	private class VibrationUser implements VibrationSystem.User {
+
+		private final ${name}Entity entity = ${name}Entity.this;
+		private final PositionSource positionSource = new EntityPositionSource(this.entity, this.entity.getEyeHeight());
+
+		@Override public PositionSource getPositionSource() {
+			return this.positionSource;
+		}
+
+		<#if data.vibrationalEvents?has_content>
+		@Override public TagKey<GameEvent> getListenableEvents() {
+			return TagKey.create(Registries.GAME_EVENT, ResourceLocation.withDefaultNamespace("${data.getModElement().getRegistryName()}_can_listen"));
+		}
+		</#if>
+
+		@Override public int getListenerRadius() {
+			<#if hasProcedure(data.vibrationSensitivityRadius)>
+				Level world = this.entity.level();
+				int x = entity.getOnPos().getX();
+                int y = entity.getOnPos().getY();

```

Comment:
Indentation here is a bit strange:

![image](https://github.com/user-attachments/assets/fee1c66b-a55e-435b-b28f-954d0ea847e6)